lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -5119,18 +5134,18 @@ Error glow::optimizeFunction(Function *F, const Backend &B,
   // instrument with profiling nodes. This must be done after lowering.
   transformForPrecisionMode(B, F, cctx);
 
-  // Fold activations before lowering to enable cases which would not fuse after
-  // lowering. This concerns particularly convolution&relu since relu will be
-  // lowered to max(0, x).
-  foldActivations(F, cctx, &B);
-
   // Lower once more, in case precision transform has introduced operators that
   // need to be lowered, e.g., Clip.
   ::glow::lower(F, cctx, &B);
 
   // Optimize the graph again now that we have a lowered representation.
   ::glow::optimize(F, cctx);
 
+  // Fold activations after optimization.
+  // Quantization nodes should be optimized before folding Activation in to
+  // Conv.
+  foldActivations(F, cctx, &B);
```

Comment:
Instead of moving this down here, maybe it could be left where it was, and we could add another call to `optimize()`? I.e. we'd have `transformForPrecisionMode()`, then `foldActivations()`, then `lower()`. Otherwise here we will only fold activations after lowering, which as the comment that was originally there states, makes pattern matching for activation folding a little more annoying.

Reply:
Sure. I moved `foldActivation` in the original position. Also, I added another `optimize()` before calling `foldActivation`.